test/e2e/apps/statefulset.go

```diff
@@ -860,14 +860,32 @@ var _ = SIGDescribe("StatefulSet", func() {
 			if err != nil {
 				framework.Failf("Failed to put scale subresource: %v", err)
 			}
-			framework.ExpectEqual(scaleResult.Spec.Replicas, int32(2))
+			framework.ExpectEqual(scaleResult.Spec.Replicas, int32(2), "")
 
 			ginkgo.By("verifying the statefulset Spec.Replicas was modified")
 			ss, err = c.AppsV1().StatefulSets(ns).Get(context.TODO(), ssName, metav1.GetOptions{})
 			if err != nil {
 				framework.Failf("Failed to get statefulset resource: %v", err)
 			}
-			framework.ExpectEqual(*(ss.Spec.Replicas), int32(2))
+			framework.ExpectEqual(*(ss.Spec.Replicas), int32(2), "")
+
+			ginkgo.By("Patch a scale subresource")
+			scale.ResourceVersion = "" // indicate the scale update should be unconditional
+			scale.Spec.Replicas = 4    // should be 2 after "UpdateScale" operation, now Patch to 4
+			ssScalePatchPayload, err := json.Marshal(autoscalingv1.Scale{
+				Spec: autoscalingv1.ScaleSpec{
+					Replicas: scale.Spec.Replicas,
+				},
+			})
+			framework.ExpectNoError(err, "Could not Marshal JSON for patch payload")
+
+			_, err = c.AppsV1().StatefulSets(ns).Patch(context.TODO(), ssName, types.StrategicMergePatchType, []byte(ssScalePatchPayload), metav1.PatchOptions{}, "scale")
+			framework.ExpectNoError(err, "Failed to patch stateful set: %v", err)
+
+			ginkgo.By("verifying the statefulset Spec.Replicas was modified")
+			ss, err = c.AppsV1().StatefulSets(ns).Get(context.TODO(), ssName, metav1.GetOptions{})
+			framework.ExpectNoError(err, "Failed to get statefulset resource: %v", err)
+			framework.ExpectEqual(*(ss.Spec.Replicas), int32(4), "statefulset should have 4 replicas")
```

Comment:
should we similar to the original status, also add ?
e2estatefulset.WaitForRunningAndReady(c, *ss.Spec.Replicas, ss)
waitForStatus(c, ss)

Reply:
It is included on line 847-848. The pattern looks to be using this only at the "create" stage of the statefulset.
Does this address your concern?

Reply:
agree its not the pattern, but extra validation makes the test more robust, that indeed after patching the scale subresource, the StatefulSet pod is ready and good to go

Reply:
Thank you for helping to make this test more solid.
I believe `WaitForRunningAndReady` is for creating new resources.
However, I think adding ` WaitForStatusReplicas` will add the value you would like to see.
I update with ` WaitForStatusReplicas` and I trust this will make the test ready to be merged.

Reply:
> agree its not the pattern, but extra validation makes the test more robust, that indeed after patching the scale subresource, the StatefulSet pod is ready and good to go

@krmayankk iirc there are other tests that ensure the scale is working as expected. The purpose of this test is to just ensure the update & patch API calls are working as expected. That additional scrutiny imo is unnecessary b/c we don't rely on the result at all and that wail will only make this test last longer. So I suggested dropping that additional wait.

Reply:
/hold
where's the e2e/conformance test that verifies stateful sets scale as expected?  as an end user I'd be surprised if the API allowed me to update the spec, but the status never reconciled